holoviews/operation/datashader.py

```diff
@@ -338,22 +360,27 @@ def _process(self, element, key=None):
             agg_kwargs['line_width'] = self.p.line_width
 
         dfdata = PandasInterface.as_dframe(data)
-        # Suppress numpy warning emitted by dask:
-        # https://github.com/dask/dask/issues/8439
-        with warnings.catch_warnings():
-            warnings.filterwarnings(
-                action='ignore', message='casting datetime64',
-                category=FutureWarning
-            )
-            agg = getattr(cvs, glyph)(dfdata, x.name, y.name, agg_fn, **agg_kwargs)
+        cvs_fn = getattr(cvs, glyph)
+
+        if sel_fn:
+            if isinstance(params["vdims"], (Dimension, str)):
+                params["vdims"] = [params["vdims"]]
+            sum_agg = ds.summary(**{str(params["vdims"][0]): agg_fn, "index": ds.where(sel_fn)})
+            agg = self._apply_datashader(dfdata, cvs_fn, sum_agg, agg_kwargs, x, y)
+            _ignore = [*params["vdims"], "index"]
```

Comment:
While it is a good default, the name "index" is a magic value. I could be wrong but couldn't this clash with a column name?